Formula/ogre.rb

```diff
@@ -0,0 +1,68 @@
+class Ogre < Formula
+  desc "Scene-oriented, flexible 3D engine"
+  homepage "http://www.ogre3d.org/"
+  url "https://github.com/OGRECave/ogre/archive/v1.10.9.tar.gz"
+  sha256 "85ba2cc6a35c67ff93a9a498af9f8f3113fd3a16e7cc43c18b9769d8bf1e9101"
+
+  needs :cxx11
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "boost" => :optional
+  depends_on "freeimage"
+  depends_on "freetype"
+  depends_on "harfbuzz"
+  depends_on "libpng"
+  depends_on "libzzip"
+  depends_on "openexr"
+  depends_on "sdl"
+  depends_on "sdl2"
+
+  def install
+    if build.with? "boost"
+      unless Tab.for_name("boost").cxx11?
+        odie "Boost build with \"--c++11\" is needed because this formula needs c++11."
+      end
+    end
+
+    # Trunk starts using "clang++ -x c" to select C compiler which breaks C++11
+    # handling using ENV.cxx11. Using "cxxflags" and "linkflags" still works.
+    ENV.cxx11
+    ENV.append "CXXFLAGS", "-std=c++11"
+    if ENV.compiler == :clang
+      ENV.append "CXXFLAGS", "-stdlib=libc++"
+      ENV.append "LINKFLAGS", "-stdlib=libc++"
+    end
+    # OGRE tends to use "/Library/Frameworks/SDL.framework" if exist. Because OGRE appends "-F/Library/Frameworks".
+    ENV.prepend "CPPFLAGS", "-I#{Formula["sdl"].opt_include}/SDL"
+
+    inreplace "CMake/Utils/OgreConfigTargets.cmake", "set(PLATFORM_NAME \"macosx\")", ""
+    inreplace "CMake/CMakeLists.txt", "set(OGRE_CMAKE_DIR \"CMake\")", "set(OGRE_CMAKE_DIR \"lib/OGRE/cmake\")"
+    inreplace ["Samples/Media/CMakeLists.txt", "CMake/InstallResources.cmake"],
+              "set(OGRE_MEDIA_PATH \"Media\")", "set(OGRE_MEDIA_PATH \"share/ogre/Media\")"
+    inreplace "CMake/InstallResources.cmake", "set(OGRE_CFG_INSTALL_PATH \"bin\")", "set(OGRE_CFG_INSTALL_PATH \"share/ogre/cfg\")"
+
+    args = std_cmake_args
+    args += %W[-DCMAKE_SKIP_BUILD_RPATH=FALSE
+               -DOGRE_FULL_RPATH=TRUE
+               -DOGRE_LIB_DIRECTORY=#{lib}
```

Comment:
I think `#{lib}` is an absolute path, but there are places in the ogre code that expect `OGRE_LIB_DIRECTORY` to not be absolute, for example the pkgconfig templates:

* https://github.com/OGRECave/ogre/blob/v1.10.9/CMake/Templates/OGRE.pc.in#L3
* https://github.com/OGRECave/ogre/blob/v1.10.9/CMake/Templates/OGRE-RTShaderSystem.pc.in#L3
* etc.

This causes the following warnings when compiling against ogre:

~~~
ld: warning: directory not found for option
 '-L/usr/local/Cellar/ogre/1.10.9/usr/local/Cellar/ogre/1.10.9/lib'
~~~

Reply:
Thank you for your advice.

Reply:
It's O.K.
```
$ brew list ogre
/usr/local/Cellar/ogre/1.10.9/bin/OgreMeshUpgrader
/usr/local/Cellar/ogre/1.10.9/bin/OgreXMLConverter
/usr/local/Cellar/ogre/1.10.9/bin/VRMLConverter
/usr/local/Cellar/ogre/1.10.9/include/OGRE/ (574 files)
/usr/local/Cellar/ogre/1.10.9/lib/libOgreBites.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreHLMS.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreMain.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreMeshLodGenerator.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreOverlay.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgrePaging.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreProperty.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreRTShaderSystem.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreTerrain.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/libOgreVolume.1.10.9.dylib
/usr/local/Cellar/ogre/1.10.9/lib/OGRE/ (23 files)
/usr/local/Cellar/ogre/1.10.9/lib/pkgconfig/ (11 files)
/usr/local/Cellar/ogre/1.10.9/lib/python2.7/ (9 files)
/usr/local/Cellar/ogre/1.10.9/lib/ (11 other files)
/usr/local/Cellar/ogre/1.10.9/share/ogre/ (944 files)
```

Reply:
I meant that the configured pkg-config files are incorrect:

~~~
$ grep 'libdir=' /usr/local/Cellar/ogre/1.10.9/lib/pkgconfig/OGRE.pc 
libdir=${prefix}//usr/local/Cellar/ogre/1.10.9/lib
~~~

Reply:
I am sorry. But `OGRE.pc` is correct now.
```
$ cat /usr/local/Cellar/ogre/1.10.9/lib/pkgconfig/OGRE.pc 
prefix=/usr/local/Cellar/ogre/1.10.9
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include
plugindir=${libdir}/OGRE

Name: OGRE
Description: Object-Oriented Graphics Rendering Engine
Version: 1.10.9
URL: http://www.ogre3d.org
Libs: -L${libdir} -lOgreMain  -lpthread
Cflags: -I${includedir} -I${includedir}/OGRE -I${includedir}/OGRE/OSX -pthread
```